src/webln/NostrWeblnProvider.ts

```diff
@@ -340,8 +337,8 @@ export class NostrWebLNProvider implements WebLNProvider, Nip07Provider {
     );
   }
 
-  keysend(args: KeysendArgs) {
-    this.checkConnected();
+  async keysend(args: KeysendArgs) {
+    await this.checkConnected();
 
     return this.executeNip47Request<SendPaymentResponse, Nip47PayResponse>(
```

Comment:
do those need to be awaited now? 

is it good to mix async/await with returned promisses?

Reply:
It's just the same as doing
```
const result = await ...;
return result;
```

Reply:
thx.